evm-contracts/src/v0.8/dev/ChainlinkClient.sol

```diff
@@ -264,8 +329,9 @@ contract ChainlinkClient {
       bytes memory
     )
   {
+    bytes4 funcSelector = (dataVersion == 2) ? oracle.requestOracleData.selector : oracle.oracleRequest.selector;
```

Comment:
I'm actually leaning towards just inlining this encoding into each of the different request types and getting rid of `encodeRequest`. Slightly less indirection, and without the ternanry it should save a little gas. I don't see much benefit in reusing this, sorry if I've gone back and forth on this.